### DISCUSSION TITLE
Static generation with multiple nested dynamic routes

### DISCUSSION THREAD

**Discussion by meandillar** [vercel/next.js#16383] [Help] [answered]
Hey everyone. I'm trying out next.js for an ecommerce store with several thousand pages which I would like to generate statically. I have come up against a few issues. Let's say I have these dynamic routes:

- `/products/[product]`
- `/collections/[collection]/products/[product]`

The [product] template in both of these scenarios is identical. A product can appear on multiple collections and I want to be able to fetch data for each product only once. The problem is that the `getStaticProps` function is at page level so each time a new static path is created for this file it fetches the data again.

I also need to fetch all collection slugs twice in this scenario when using `getStaticPaths` - once for dynamically creating the [collection] paths, and again in the collection [product] file in order to find the products on a particular collection - this issue would be solved if I could access the dynamic collection param from within `getStaticPaths` but it seems like that's not an option.

One last problem is that I'd like to be able to run the api requests in series across all statically generated pages. Currently I can only do this per file and I'm finding it hard to manage api rate limits because requests across multiple pages fire all kick off at once.

I guess what would really suit my use case is to be able to make all api requests for static pages and trigger static page generation from one file. Is anything like this possible?

Cheers in advance!

**Comment by meandillar** [comment, marked as answer]
Never mind, I just discovered `fallback: 'unstable_blocking'` yay!

**Reply by tylorkolbeck** [reply]
Hello! I am trying to setup the same scenario. Do you by chance have a repo that I could reference?

**Reply by meandillar** [reply]
Hey I don't have a repo for you to look at but I've ended up with something like this for dynamic routes where I request my top 100 product paths to create static pages at build time, and the I use `fallback: true` for the rest, which will mean that these pages are in a loading state when first visited.

```
export async function getStaticPaths() {
  const topProductPaths = await someFunctionToGetPaths()
  
  return {
    paths: topProductPaths,
    fallback: true
  }
}
```
